### ISSUE TITLE
CID whitelist

### ISSUE THREAD

**Issue by liangzhenhou** [STOmics/SAW#136]
Hi, 
could you tell me that which file in Stereo-seq data is or contains CID whitelist? 
And when I get the CID barcode, how could I know the position in the chip?
Thank you !

**Comment by Jinghao-Chen** [comment]
Hi, did you need the fastq that could be mapped to the Stereo-seq chip? If so, you need to add `validCidFq` to the last line of the *.bcPara file in 00.mapping, and then rerun the mapping step to obtain the fastq file containing valid CID reads, CID sequence and their positions in the chip as shown below(from Documents/UserManual/Stereo-seq_Analysis_Workflow_User_Manual_A9.pdf Page 23).
![image](https://github.com/user-attachments/assets/d02f9e8b-368c-4d33-98e9-a6207bda7cc5)

In addition, if you want to obtain all CID sequences and their positions on the chip, it is recommended to use ST_BarcodeMap. The conversion method is 'mask format change' at the bottom of the README.
https://github.com/STOmics/ST_BarcodeMap

**Comment by liangzhenhou** [comment]
Thanks for your reply!
I have the fastq files with CID sequences, which need to be sequencing error corrected to whitelist CID.
I already have the procedure of the error correction. 
Do you mean that the mask.h5 file contains the CID barcode whitelist and their positions on the chip?

**Comment by Jinghao-Chen** [comment]
> Thanks for your reply! I have the fastq files with CID sequences, which need to be sequencing error corrected to whitelist CID. I already have the procedure of the error correction. Do you mean that the mask.h5 file contains the CID barcode whitelist and their positions on the chip?

Yes, SN.barcodeToPos.h5 contains the CID and their positions. Inside is a matrix, and the value of the matrix[y, x, 0] is the transcoded CID sequence. ST_barcodemap could effectively help convert it to a txt file whose format is `sequence x y`

**Comment by an-altosian** [comment]
Thank you for the discussions. I have a follow question: I saw from the user manual that the CID is 25 bases; UMI is 10 bases; and UMIs are on read1s. however, my Stereo-seq data has only 31 bases in reads1, and 85 bases in reads2. Could you please provide some advice on how to locate the CBs and UMIs in the FASTQ records? Thank you very much.

**Comment by Jinghao-Chen** [comment]
@an-altosian Hi, this sequencing data looks like the data from the Stereo-seq FFPE Early Access Kit, with 25bp CID + 6bp UMI on read1 and mRNA on read2. If so, you need to use `docker pull stomics/saw_beta:v6.1.0a4` for analysis and adapt to the specific analysis process. If not, you may need to contact technical support to obtain library structure information. And then refer to the user manual to specify `barcodeStart, barcodeLen, umiStart, umiLen` by editing the .bcPara file
